spec/pagerduty/legacy_spec.rb

```diff
@@ -325,22 +311,24 @@
           }
         end
 
-        context "with incident_key option" do
-          When(:trigger) {
-            incident.trigger(
-              "description",
-              incident_key: "method param incident_key",
-            )
-          }
-          Then {
-            expect(transport).to have_received(:send_payload).with(
-              incident_key: "method param incident_key",
-              service_key:  "a-test-service-key",
-              event_type:   "trigger",
-              description:  "description",
-            )
-          }
-        end
+        # This is the only gem version 2 spec that no-longer works.
+        #
+        # context "with incident_key option" do
+        #   When(:trigger) {
+        #     incident.trigger(
+        #       "description",
+        #       incident_key: "method param incident_key",
+        #     )
+        #   }
+        #   Then {
+        #     expect(transport).to have_received(:send_payload).with(
+        #       incident_key: "method param incident_key",
+        #       service_key:  "a-test-service-key",
+        #       event_type:   "trigger",
+        #       description:  "description",
+        #     )
+        #   }
+        # end
```

Comment:
This is the only test that doesn't pass. In this scenario an `ArgumentError` is now raised.

  ```ruby
  incident1 = pagerduty.trigger('first incident', incident_key: 'one') # this'll work fine
  incident2 = incident1.trigger('second incident', incident_key: 'two') # this no-longer works.
  ```

  The difference is in the object we're calling the method on. ie. It's ok to pass an `incident_key` in the `pagerduty.trigger` method. But not ok to pass it into the `incident.trigger` method.

  Instead always use the `pagerduty` object when triggering new incidents (with new incident keys).

  This works:

  ```ruby
  incident1 = pagerduty.trigger('first incident', incident_key: 'one')
  incident2 = pagerduty.trigger('second incident', incident_key: 'two')
  ```

  And this is even better:

  ```ruby
  incident1 = pagerduty.get_incident('one').trigger('first incident')
  incident2 = pagerduty.get_incident('two').trigger('second incident')
  ```

Reply:
Is it worth adjusting the specs to demonstrate the new behaviour?

Reply:
This spec file (`legacy_spec`), is here to demonstrate compatibility with the current version of this gem. I don't think we want to include anything that wasn't a part of the original behaviour.

FYI: this behaviour is covered in the `events_api_v1_spec` ([see here](https://github.com/envato/pagerduty/pull/64/files#diff-9077cb9fdc5ce6c5d468d7b643d1bf1cR318-R321)).

Reply:
💯